### PR TITLE
chore: remove unneeded usage of `pub` throughout gateway code

### PR DESCRIPTION
**Pull request by tvolk131**

### CODE REVIEW

File: gateway/ln-gateway/src/lib.rs [hunk 1]
```diff
@@ -1165,7 +1165,7 @@ impl Gateway {
     /// - `GatewayConfiguration` is read from the database.
     /// - All cli or environment variables are set such that we can create a
     ///   `GatewayConfiguration`
-    pub async fn get_gateway_configuration(&self) -> Option<GatewayConfiguration> {
+    async fn get_gateway_configuration(&self) -> Option<GatewayConfiguration> {
```
Comment (elsirion) [inline gateway/ln-gateway/src/lib.rs:1168]: Is this something you might want to call if you instantiate the gateway yourself as part of a custom binary?

File: gateway/ln-gateway/src/state_machine/mod.rs [hunk 2]
```diff
@@ -240,7 +239,7 @@ impl ClientModule for GatewayClientModule {
 }
 
 impl GatewayClientModule {
-    pub fn to_gateway_registration_info(
+    fn to_gateway_registration_info(
```
Comment (elsirion) [inline gateway/ln-gateway/src/state_machine/mod.rs:242]: Maybe useful for integrators

**Review by elsirion — APPROVED** [review]
I think in this case it's ok to reduce visibility, but @douglaz made a good point yesterday that making things pub again when needed is a lengthy process. So maybe for now we only want to make things private where it's necessary for guaranteeing invariants.

**Comment by elsirion** [comment]
@douglaz do you think this PR is worth it? If so it needs a rebase.

**Comment by tvolk131** [comment]
Rebased!

**Review by elsirion — APPROVED** [review]

File: gateway/ln-gateway/src/lib.rs (hunk 1 above)
Comment (m1sterc001guy) [reply gateway/ln-gateway/src/lib.rs:1206]: I'd say yes, but right now I don't see us supporting custom binaries for gateways

**Review by m1sterc001guy — APPROVED** [review]
Send it

**Pull request merged**